### PR TITLE
Interpret build arguments as Dep_conf.t

### PR DESCRIPTION
**Pull request by nojb**
As suggested by @rgrinberg, this is part of #2417 for easier reviewing.

A cmdliner converter is added for `Dune_file.Dep_conf.t` and used to interpret arguments to `build` command.

In principle there shouldn't be any observable difference with respect to existing functionality, even though new syntax is accepted for build arguments (see the test).

### CODE REVIEW

File: bin/arg.ml [hunk 1]
```diff
@@ -25,3 +25,94 @@ let path = Path.conv
...
+      `Ok (alias s)
+    | None ->
+      match
+        Dune_lang.Decoder.parse dep_parser Univ_map.empty
+          (Dune_lang.parse_string ~fname:"command line"
+             ~mode:Dune_lang.Parser.Mode.Single s)
+      with
+      | x -> `Ok x
+      | exception (User_error.E msg) ->
+        `Error (User_message.to_string msg)
```
Comment (rgrinberg) [inline bin/arg.ml]: As mentioned previously, I'm not entirely sure this prints the errors correctly.

Comment (nojb) [reply bin/arg.ml]: I added a test showing the kind of errors that come out.

Comment (rgrinberg) [reply bin/arg.ml]: Yeah, it doesn't look all that good but I suppose it's good enough for now.

**Comment by rgrinberg** [comment]
Ping @nojb

@diml could you glance at the `User_error` issue. I'm not sure we can fix this in the short term, but it would be good to have a plan.

**Comment by nojb** [comment]
@rgrinberg thanks for the ping, this is ready to go as far as I know, but I was waiting in case @diml had any comments, since he had been reviewing #2417.

(Also, I have spotty connectivity at the moment. I will have to wait until I get back tomorrow night to bring the PR up to date and --if all is well-- merge).

**Comment by diml** [comment]
That seems like a pain to change as we raise `User_error` directly at the source everywhere. We do this matching in only one place in the code base, so it doesn't seem worth changing everything just for that. At the end of the day, the produced error messages look fine.

**Comment by nojb** [comment]
Thanks for the reviews! I rebased the PR and will merge it once it goes through CI.

**Pull request merged**